### PR TITLE
Datadog exporter: migrate to intake v2

### PR DESCRIPTION
**Pull request by nachoBonafonte**
Updated the endpoints and the format.
It support both API key and client keys in the same configuration parameter
Added support for compressing the payloads

### CODE REVIEW

**Review by SergeyKanzhelev — APPROVED** [review]
just glanced over, lgtm

File: Sources/Exporters/DatadogExporter/ExporterConfiguration.swift [hunk 1]
```diff
@@ -37,15 +38,15 @@ public struct ExporterConfiguration {
     /// Option to add a custom prefix to all the metrics sent by the exporter
     var metricsNamePrefix: String?
 
-    public init(serviceName: String, resource: String, applicationName: String, applicationVersion: String, environment: String, clientToken: String?, apiKey: String?, endpoint: Endpoint, uploadCondition: @escaping () -> Bool, performancePreset: PerformancePreset = .default, exportUnsampledSpans: Bool = true, exportUnsampledLogs: Bool = true, hostName: String? = nil, metricsNamePrefix: String? = "otel") {
+    public init(serviceName: String, resource: String, applicationName: String, applicationVersion: String, environment: String, apiKey: String, endpoint: Endpoint, source: String = "otel-swift-exporter", uploadCondition: @escaping () -> Bool, performancePreset: PerformancePreset = .default, exportUnsampledSpans: Bool = true, exportUnsampledLogs: Bool = true, hostName: String? = nil, metricsNamePrefix: String? = "otel") {
```
Comment (ncreated) [inline Sources/Exporters/DatadogExporter/ExporterConfiguration.swift:41]: Wasn't `source` set to `"ios"` before? 
I'm not exactly sure about the impact of using `"otel-swift-exporter"` 💭 - I think this won't install default "iOS" processing pipelines (in Logs) when the first data comes in.

File: Sources/Exporters/DatadogExporter/Logs/LogsExporter.swift [hunk 2]
```diff
@@ -39,7 +35,7 @@ internal class LogsExporter {
             dateProvider: SystemDateProvider()
         )
 
-        let dataFormat = DataFormat(prefix: "", suffix: "", separator: "\n")
+        let dataFormat = DataFormat(prefix: "[", suffix: "]", separator: ",")
```
Comment (ncreated) [inline Sources/Exporters/DatadogExporter/Logs/LogsExporter.swift:38]: 👍

File: Sources/Exporters/DatadogExporter/Upload/DataUploadStatus.swift [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    let needsRetry: Bool
+
+    // MARK: - Debug Info
+
+    /// Upload status description printed to the console if SDK `.debug` verbosity is enabled.
+    let userDebugDescription: String
+
+    /// An optional error printed to the console if SDK `.error` (or lower) verbosity is enabled.
+    /// It is meant to indicate user action that must be taken to fix the upload issue (e.g. if the client token is invalid, it needs to be fixed).
+    let userErrorMessage: String?
```
Comment (ncreated) [inline Sources/Exporters/DatadogExporter/Upload/DataUploadStatus.swift:63]: I don't see if we make use of these two in `DataUploadWorker`. In `dd-sdk-ios` we print the `userDebugDescription` to the user (on upload failure) and send `userErrorMessage` for internal monitoring. I think we can remove it from here and cleanup.

**Review by ncreated — APPROVED** [review]
Looks good 👍, I left few clean-up comments.

File: Sources/Exporters/DatadogExporter/ExporterConfiguration.swift (hunk 1 above)
Comment (nachoBonafonte) [reply Sources/Exporters/DatadogExporter/ExporterConfiguration.swift:41]: Oh, I was not aware of that. I will change it back

File: Sources/Exporters/DatadogExporter/Upload/RequestBuilder.swift [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        }
+
+        self.url = urlComponents?.url ?? url
+        self.precomputedHeaders = precomputedHeaders
+        self.computedHeaders = computedHeaders
+    }
+
+    /// Creates `URLRequest` for uploading given `data` to Datadog.
+    /// - Parameter data: data to be uploaded
+    /// - Returns: the `URLRequest` object and `DD-REQUEST-ID` header value (for debugging).
```
[the remark this answers is not in the document]
Comment (nachoBonafonte) [reply Sources/Exporters/DatadogExporter/Upload/RequestBuilder.swift]: Removed

File: Sources/Exporters/DatadogExporter/Upload/DataUploadStatus.swift (hunk 3 above)
Comment (nachoBonafonte) [reply Sources/Exporters/DatadogExporter/Upload/DataUploadStatus.swift]: Cleaned up, thanks for your review

**Pull request merged**